gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -85,64 +77,39 @@ public class Orchestrator implements SpecCatalogListener, Instrumentable {
   protected final SpecCompiler specCompiler;
   protected final TopologyCatalog topologyCatalog;
-  protected final DagManager dagManager;
   private final JobStatusRetriever jobStatusRetriever;
 
   protected final MetricContext metricContext;
 
   protected final EventSubmitter eventSubmitter;
-  private final boolean isFlowConcurrencyEnabled;
   @Getter
   private Meter flowOrchestrationSuccessFulMeter;
   @Getter
   private Meter flowOrchestrationFailedMeter;
   @Getter
   private Timer flowOrchestrationTimer;
-  private Counter flowFailedForwardToDagManagerCounter;
-  @Setter
-  private FlowStatusGenerator flowStatusGenerator;
-
-  private UserQuotaManager quotaManager;
-  private final FlowCompilationValidationHelper flowCompilationValidationHelper;
-  private Optional<FlowLaunchHandler> flowLaunchHandler;
-  private Optional<FlowCatalog> flowCatalog;
+  private final FlowLaunchHandler flowLaunchHandler;
   @Getter
   private final SharedFlowMetricsSingleton sharedFlowMetricsSingleton;
 
   @Inject
-  public Orchestrator(Config config, TopologyCatalog topologyCatalog, DagManager dagManager,
-      Optional<Logger> log, FlowStatusGenerator flowStatusGenerator, Optional<FlowLaunchHandler> flowLaunchHandler,
-      SharedFlowMetricsSingleton sharedFlowMetricsSingleton, Optional<FlowCatalog> flowCatalog,
-      Optional<DagManagementStateStore> dagManagementStateStore,
+  public Orchestrator(Config config, TopologyCatalog topologyCatalog, Optional<Logger> log, FlowLaunchHandler flowLaunchHandler,
+      SharedFlowMetricsSingleton sharedFlowMetricsSingleton, DagManagementStateStore dagManagementStateStore,
       FlowCompilationValidationHelper flowCompilationValidationHelper, JobStatusRetriever jobStatusRetriever) throws IOException {
     _log = log.isPresent() ? log.get() : LoggerFactory.getLogger(getClass());
     this.topologyCatalog = topologyCatalog;
-    this.dagManager = dagManager;
-    this.flowStatusGenerator = flowStatusGenerator;
     this.flowLaunchHandler = flowLaunchHandler;
     this.sharedFlowMetricsSingleton = sharedFlowMetricsSingleton;
-    this.flowCatalog = flowCatalog;
     this.jobStatusRetriever = jobStatusRetriever;
-    this.flowCompilationValidationHelper = flowCompilationValidationHelper;
     this.specCompiler = flowCompilationValidationHelper.getSpecCompiler();
-    //At this point, the TopologySpecMap is initialized by the SpecCompiler. Pass the TopologySpecMap to the DagManager.
-    this.dagManager.setTopologySpecMap(getSpecCompiler().getTopologySpecMap());
-    if (dagManagementStateStore.isPresent()) {
-      ((MySqlDagManagementStateStore) dagManagementStateStore.get()).setTopologySpecMap(getSpecCompiler().getTopologySpecMap());
-    }
+    //At this point, the TopologySpecMap is initialized by the SpecCompiler. Pass the TopologySpecMap to the DagManagementStateStore.
+    ((MySqlDagManagementStateStore) dagManagementStateStore).setTopologySpecMap(getSpecCompiler().getTopologySpecMap());
```

Comment:
seeing this cast makes me wonder:
a. does `setTopologySpecMap` belong in the `DagManagementStateStore` interface?
b. OR, perhaps even better, should it be a ctor param to the `MysqlDagManagementStateStore`?  I realize there are timing interdependencies to coordinate, but couldn't those be handled before `MysqlDagManagementStateStore::start` is to be called?

Reply:
added [GOBBLIN-2056](https://issues.apache.org/jira/browse/GOBBLIN-2056)